great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1035,6 +1191,151 @@ def _sample_using_md5(
+        column_name. This data can be used to build BatchIdentifiers.
+
+        Args:
+            table_name: table to split.
+            column_name: column in table to use in determining split.
+
+        Returns:
+            List of dicts of the form [{column_name: {"year": 2022, "month": 4, "day": 14}}]
+        """
+        return self.get_data_for_batch_identifiers_for_split_on_date_parts(
+            table_name=table_name,
+            column_name=column_name,
+            date_parts=[DatePart.YEAR, DatePart.MONTH, DatePart.DAY],
+        )
+
+    def get_data_for_batch_identifiers_for_split_on_date_parts(
+        self,
+        table_name: str,
+        column_name: str,
+        date_parts: Union[List[DatePart], List[str]],
+    ) -> List[dict]:
+        """Build batch_identifiers from a column split on a list of date parts.
+
+        This method builds a query to select the unique date_parts from the
+        column_name. This data can be used to build BatchIdentifiers.
+
+        Args:
+            table_name: table to split.
+            column_name: column in table to use in determining split.
+            date_parts: part of the date to be used for splitting e.g.
+                DatePart.DAY or the case-insensitive string representation "day"
+
+        Returns:
+            List of dicts of the form [{column_name: {date_part_name: date_part_value}}]
+        """
+        if len(date_parts) == 0:
+            raise ge_exceptions.InvalidConfigError(
+                "date_parts are required when using split_on_date_parts."
+            )
+
+        date_parts: List[DatePart] = [
+            DatePart(date_part.lower()) if isinstance(date_part, str) else date_part
+            for date_part in date_parts
+        ]
+
+        # NOTE: AJB 20220414 concatenating to find distinct values to support all dialects.
+        # There are more performant dialect-specific methods that can be implemented in
+        # future improvements.
+        if len(date_parts) == 1:
+            # MSSql does not accept single item concatenation
+            concat_clause = [
+                sa.func.distinct(
+                    sa.func.extract(date_parts[0].value, sa.column(column_name)).label(
+                        date_parts[0].value
+                    )
+                ).label("concat_distinct_values")
+            ]
+        else:
+            concat_clause = [
+                sa.func.distinct(
+                    sa.func.concat(
+                        *[
+                            (
+                                sa.func.extract(date_part.value, sa.column(column_name))
+                            ).label(date_part.value)
+                            for date_part in date_parts
+                        ]
+                    )
+                ).label("concat_distinct_values"),
+            ]
+
+        split_query: "sa.sql.expression.Select" = sa.select(
```

Comment:
@anthonyburdi Are the quoted types required?  I am guessing that we can import that required types.  But if required, then we would need the ` # noqa: F821` in that line as well.  Thanks.

Reply:
Good call - I'll add the `# noqa: F821`. I left these hints in quotes because there is no guarantee that the user has `sqlalchemy` installed and I didn't want there to be an issue with trying to import it for these type hints.

Reply:
@anthonyburdi We have a way to import with error handling -- it is designed exactly for the situations where the user is given SQLAlchemy when needed, and not otherwise.  I feel that we should adhere to that code pattern -- it is a just a little bit more work, but it has been demonstrated to be effective.  Thanks!